[stanleytangzh] iP

stanleytangzh opened the pull request:
# Stan
“The best way to predict the future is to invent it.” – Alan Kay 

Stan is here to make your task management effortless and efficient. With Stan, you can:

* Organize your tasks
* Track your deadlines
* Manage your events with ease
* Here's how you can get started:

1. Download Stan from the [release page](https://github.com/stanleytangzh/ip/releases).
2. Double-click on the stan.jar file.
3. Start adding your tasks with simple commands.
4. Let Stan do the heavy lifting for you! 😎

Features:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Event scheduling
- [ ] Reminders (coming soon)

Stan is fast, easy to use, and super flexible. Plus, it's **FREE**! If you’re a Java developer, Stan is also a great project to practice your coding skills.

Here's a glimpse of the `main` method that powers Stan:
```java
public class Stan {
    public static void main(String[] args) {
        new Stan("data/stan.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

stanleytangzh committed:
renamed chabot

stanleytangzh committed:
implemented an initial skeletal version that simply greets and user and exits.

stanleytangzh committed:
echo user command and exit upon bye command.

stanleytangzh committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

stanleytangzh committed:
Mark as Done

stanleytangzh committed:
ToDos, Events, Deadlines, A-TextUiTesting

stanleytangzh committed:
Handle Errors

stanleytangzh committed:
Added enum command type

stanleytangzh committed:
Add delete task support and case statement with commandType

stanleytangzh committed:
make Task abstract

stanleytangzh committed:
create greet and getList methods

stanleytangzh committed:
Add exception classes and improve error handling

stanleytangzh committed:
Add text ui test cases

stanleytangzh committed:
Implement the Storage class to read and save tasks automatically whenever the task list changes

stanleytangzh committed:
Add date and time functionality in deadline and event objects

stanleytangzh committed:
Make the code more OOP by extracting relevant classes

stanleytangzh committed:
Divide classes into packages

stanleytangzh committed:
Ignore stan.txt

stanleytangzh committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Ignore stan.txt
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

stanleytangzh committed:
Add JUnit tests for TaskList class

stanleytangzh committed:
Add JUnit tests for Deadline, Event and Todo classes

stanleytangzh committed:
Modify build.gradle setting for creating JAR file

stanleytangzh committed:
Modify build.gradle setting and rebuild Jar file

stanleytangzh committed:
Add JavaDoc comments to the code

stanleytangzh committed:
Tweak the code to comply with the coding standard

stanleytangzh committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc comments to the code

stanleytangzh committed:
Add functionality of finding a task by searching for a keyword

stanleytangzh committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Tweak the code to comply with the coding standard

# Conflicts:
#	src/main/java/stan/Parser.java

stanleytangzh committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add functionality of finding a task by searching for a keyword

# Conflicts:
#	src/main/java/stan/Parser.java
#	src/main/java/stan/TaskList.java
#	src/main/java/stan/Ui.java
#	src/main/java/stan/tasks/Task.java

xhamyo commented on src/main/java/stan/CommandType.java:
```diff
@@ -0,0 +1,5 @@
+package stan;
+
+public enum CommandType {
+    BYE, LIST, TODO, EVENT, DEADLINE, MARK, UNMARK, DELETE, FIND
```
>Class/enum names must be nouns and written in PascalCase.

I think the names should be Bye, List, ToDo etc. instead of full caps.

xhamyo commented on src/main/java/stan/Storage.java:
```diff
@@ -0,0 +1,139 @@
+            return tasks;
+        }
+
+        try (Scanner sc = new Scanner(file)) {
```
Maybe need to close the scanner before the method ends?

xhamyo commented on src/test/java/stan/TaskListTest.java:
```diff
@@ -0,0 +1,62 @@
+package stan;
```
Maybe can add blank line after package.

xhamyo commented on src/test/java/stan/MockTask.java:
```diff
@@ -0,0 +1,20 @@
+package stan;
```
Maybe can add blank line after package.

xhamyo commented on src/test/java/stan/MockTask.java:
```diff
@@ -0,0 +1,20 @@
+package stan;
+import stan.tasks.Task;
+public class MockTask extends Task {
```
Maybe can add blank line after import.

xhamyo reviewed (commented):
LGTM. Just some nits to fix.

stanleytangzh committed:
Add a GUI to the chatbot using the JavaFX technology

stanleytangzh committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add a GUI to the chatbot using the JavaFX technology

Droas590 commented on src/main/java/stan/Storage.java:
```diff
@@ -0,0 +1,139 @@
+     * @throws IOException If an I/O error occurs.
+     */
+    private static File createFile(String dirPath, String filePath) throws IOException {
+        File directory = new File(dirPath);
```
Just a nit, maybe put fileDirectory will allow easier understading.

Droas590 commented on src/main/java/stan/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * @param keyword The keyword to search for.
+     * @return A list of tasks that match the keyword.
+     */
+    public List<Task> findTasks(String keyword) {
```
Maybe can add explaining element in name, eg: findTasksByString

Droas590 commented on src/main/java/stan/exceptions/StanInvalidCommandException.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * Represents an exception thrown when an invalid command is entered by the user.
+ */
+public class StanInvalidCommandException extends StanException {
```
I like the exceptions naming.

Droas590 reviewed (commented):
LGTM. The namings are easily understandable. Just some nits.

stanleytangzh committed:
Add checkstyle

stanleytangzh committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Add checkstyle

stanleytangzh committed:
A-Assertions: Add runtime checks with assertions

Certain assumptions about the program state are implicit in the code
but remain unchecked.

Java assertions provide a way to validate these assumptions at runtime,
helping to detect potential issues early during development.

Let's introduce assertions to ensure key assumptions about inputs
and program state are explicitly verified at runtime.

stanleytangzh committed:
Merge pull request #1 from stanleytangzh/branch-A-Assertions

A-Assertions: Add runtime checks with assertions

stanleytangzh committed:
A-CodeQuality: Refactor to enhance code quality

Some methods in the codebase lack adherence to code quality, making them harder to follow and maintain.

Refactoring can help improve both code structure and readability.

This commit refactors the Parser and TaskList classes to:
* Enhance readability and ensure proper error handling in the Parser class.
* Eliminate code duplication by using constructor chaining in the TaskList class constructors.

stanleytangzh committed:
Merge branch 'master' into branch-A-CodeQuality

stanleytangzh committed:
Merge pull request #2 from stanleytangzh/branch-A-CodeQuality

A-CodeQuality: Refactor to enhance code quality

stanleytangzh committed:
A-Streams: Refactor findTasks method to use Java Streams

The current findTasks method manually iterates over the task list
to filter tasks based on the keyword, making the code less concise.

Using Java Streams simplifies the filtering process and improves
code readability while maintaining the same functionality.

This refactor:
* Uses the stream API to process tasks and filter them based on the keyword.
* Utilizes Collectors.toList() to gather filtered tasks into a List.
* Ensures assertions are preserved to validate input and task integrity.

This change enhances the overall clarity of the method and leverages modern Java features for improved maintainability.

stanleytangzh committed:
Merge branch 'branch-A-Streams'

* branch-A-Streams:
  A-Streams: Refactor findTasks method to use Java Streams

stanleytangzh committed:
A-CI: set up CI

There is no Continuous integration in the project,
so we have to build the project manually each time.

With CI, integration, building, and testing can
happen automatically after each code change.

Let's use GitHub Actions to set up CI.

stanleytangzh committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  A-CI: set up CI

stanleytangzh committed:
Add CI related improvement

stanleytangzh committed:
Add CI related improvement

stanleytangzh committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Add CI related improvement
  Add CI related improvement

stanleytangzh committed:
Fix import order

stanleytangzh committed:
Make the command syntax more flexible by providing shorter aliases for keywords

stanleytangzh committed:
Merge branch 'branch-C-FriendlierSyntax'

* branch-C-FriendlierSyntax:
  Make the command syntax more flexible by providing shorter aliases for keywords

stanleytangzh committed:
Update runtest.sh file

stanleytangzh committed:
Modify runtest.sh to fix build bug

stanleytangzh committed:
Modify runtest.bat to fix bug

stanleytangzh committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Modify runtest.bat to fix bug

stanleytangzh committed:
Set the main window dimensions

stanleytangzh committed:
Improve dialogbox and background

stanleytangzh committed:
Fix reply label direction

stanleytangzh committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Fix reply label direction
  Improve dialogbox and background
  Set the main window dimensions

stanleytangzh committed:
Add personality to chat bot

stanleytangzh committed:
Merge branch 'branch-A-Personality'

* branch-A-Personality:
  Add personality to chat bot

stanleytangzh committed:
Add mark and unmark error message

stanleytangzh committed:
Merge branch 'branch-A-MoreErrorHandling'

* branch-A-MoreErrorHandling:
  Add mark and unmark error message

stanleytangzh committed:
Add more JUnit tests

stanleytangzh committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add more JUnit tests

stanleytangzh committed:
Add more testing and fix storage error

stanleytangzh committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add more testing and fix storage error

stanleytangzh committed:
Add UTF-8 encoding

stanleytangzh committed:
Merge branch 'branch-A-MoreTesting'

* branch-A-MoreTesting:
  Add UTF-8 encoding

stanleytangzh committed:
Add a user guide

stanleytangzh committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add a user guide

stanleytangzh committed:
Modify the user guide

stanleytangzh committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Modify the user guide
